scipy/stats/_distn_infrastructure.py

```diff
@@ -900,7 +901,7 @@ def _rvs(self, *args):
         # generated.
 
         ## Use basic inverse cdf algorithm for RV generation as default.
-        U = self._random_state.random_sample(self._size)
+        U = self._random_state.uniform(size=self._size)
```

Comment:
I realize this is not intended to change anything numerically and (presumably) the outputs are unchanged. Are time/benchmarks impacted at all? I'd expect not to be and not shouldn't be changed in this PR if impacted but wanted to call out to make sure we know whether any change.

Reply:
`RandomState.{uniform, random_sample}` give the same stream of floats if setup with the same seed, so existing behaviour is unchanged. I changed the usage to `uniform` because both `Generator` and `RandomState` have a `uniform` method.

```
>>> from scipy._lib._util import check_random_state
>>> rng = check_random_state(9845)
>>> print(rng.uniform(size=5))
[0.0140971  0.71189422 0.70789811 0.9087384  0.23947253]
>>> rng = check_random_state(9845)
>>> print(rng.random_sample(size=5))
[0.0140971  0.71189422 0.70789811 0.9087384  0.23947253]
```

Here are some timings:
```
>>> from numpy.random import RandomState
>>> rs = RandomState(9845)

>>> %timeit rs.uniform(size=1000)
13.8 µs ± 314 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
>>> %timeit rs.uniform(size=1)
3.15 µs ± 72.3 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

>>> %timeit rs.random_sample(size=1000)
10.3 µs ± 394 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
>>> %timeit rs.random_sample(size=1)
1.26 µs ± 60 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

>>> gen = np.random.default_rng()
>>> %timeit gen.uniform(size=1000)
8.42 µs ± 135 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
>>> %timeit gen.uniform(size=1)
3.27 µs ± 98.1 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Only requesting a single float will incur a (small) performance hit. When one asks for lots of floats then a `Generator` is faster. The `RandomState.uniform` method is not as fast as `RandomState.random_sample`.

Reply:
cool, I'd imagine the single float request is less common than several so this may even improve speed w/new `Generator` in some cases-great. Thanks for putting the timings together.